src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,98 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
 public class WorkWithFile {
+
+    private static final String ERROR_MSG = "Can't read from file.";
+    private static final String SEPARATOR_REGEX = ",";
+    private static final int VALUE_INDEX = 0;
+    private static final int KEY_INDEX = 1;
+    private static final String RESULT_LINE = "result,%s";
+    private static final String SUPPLY_ALWAYS_FIRST = "supply";
+    private List<String[]> linesArrays;
+    private String[] values;
+    private int[] keys;
+    private StringBuilder finalResults;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        linesArrays = new ArrayList<>();
+        finalResults = new StringBuilder();
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            String value = reader.readLine();
+
+            while (value != null) {
+                linesArrays.add(value.split(SEPARATOR_REGEX));
+                value = reader.readLine();
+            }
+            values = mergeValues(linesArrays);
+            keys = mergeKeys(linesArrays, values);
+            finalResults = calculateResults(values, keys);
+        } catch (IOException e) {
+            throw new RuntimeException(ERROR_MSG + e);
+        }
+
+        writeToFile(finalResults, toFileName);
```

Comment:
Let's implement getStatistic() method this way:
```suggestion
       String dataFromFile = readFile(fromFileName);
       String report = createReport(dataFromFile);
       writeToFile(report, toFileName);
```